Source/CrossingDetector/CrossingDetector.cpp

```diff
@@ -157,7 +170,7 @@ void CrossingDetector::process(AudioSampleBuffer& continuousBuffer)
 
         //in binary arrays move previous values back and make space for new value
         //include counter for above threshold
-        if(pastBinary[0] == 1)
+        if(pastSpan >= 1 && pastBinary[0] == 1)
```

Comment:
I added this `pastSpan >= 1` condition in each place where we're changing `pastCounter` or `futureCounter` to get rid of a bug when `pastSpan` or `futureSpan` is 0 and these lines either access outside the array or assume that samples on either side of the crossing are part of the past or future sample voting span.